### PR TITLE
Allow Private Worker Pool for Cloud Functions Cloud Builds.

### PR DESCRIPTION
**Pull request by gleichda**
Allow Private Worker Pool for Cloud Functions Cloud Builds.

Fixes https://github.com/hashicorp/terraform-provider-google/issues/10897

If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudfunctions: Added `build_worker_pool` to `google_cloudfunctions_function`
```

### CODE REVIEW

**Comment by gleichda** [comment]
Hey @melinath any chance you can take a look on that this week? We're waiting to get that merged

**Comment by melinath** [comment]
apologies, I'm behind on my review queue and will get to this as soon as I can.

File: mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb [hunk 1]
```diff
@@ -246,6 +246,42 @@ func TestAccCloudFunctionsFunction_update(t *testing.T) {
...
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckCloudFunctionsFunctionDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudFunctionsFunction_buildworkerpool(functionName, bucketName, zipFilePath),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCloudFunctionsFunctionExists(
+						t, funcResourceName, &function),
+					// TODO
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb:270]: what is this TODO?

**Review by melinath — CHANGES_REQUESTED** [review]
this looks pretty straightforward & should be fine. Please update the release note according to the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) and update the resource's docs. I also have one question below.

File: mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb (hunk 1 above)
Comment (gleichda) [reply mmv1/third_party/terraform/tests/resource_cloudfunctions_function_test.go.erb]: This was my hint to not forget to implement the test. Fixed now

**Review by melinath — APPROVED** [review]

**Pull request merged**